src/Compilers/Test/Utilities/CSharp/FunctionPointerUtilities.cs

```diff
@@ -128,6 +130,42 @@ static void verifyParameter(ParameterSymbol symbol, MethodSymbol containing)
             }
         }
 
+        public static void VerifyPublicFunctionPointerGetTypeInfo(SemanticModel model, SyntaxNode syntax, bool useConvertedType,
+            Action<ITypeSymbol> returnTypeVerifier,
+            params Action<ITypeSymbol>[] paramVerifiers)
+        {
+            var typeInfo = model.GetTypeInfo(syntax);
+            var typeSymbol = (IFunctionPointerTypeSymbol)(useConvertedType ? typeInfo.ConvertedType : typeInfo.Type)!;
+
+            if (!useConvertedType)
+            {
+                Assert.Equal(typeInfo.Type, typeInfo.ConvertedType, SymbolEqualityComparer.IncludeNullability);
+            }
+
+            CommonVerifyFunctionPointer((FunctionPointerTypeSymbol)((CSharp.Symbols.PublicModel.FunctionPointerTypeSymbol)typeSymbol).UnderlyingTypeSymbol);
+
+            returnTypeVerifier(typeSymbol.Signature.ReturnType);
+
+            Assert.Equal(paramVerifiers.Length, typeSymbol.Signature.Parameters.Length);
+
+            foreach (var (verifier, parameter) in paramVerifiers.Zip(typeSymbol.Signature.Parameters, (v, p) => (v, p)))
+            {
+                Assert.Empty(parameter.Name);
+                verifier(parameter.Type);
+            }
+
+            if (syntax is FunctionPointerTypeSyntax { Parameters: var parameterSyntaxes })
+            {
+                Assert.Equal(paramVerifiers.Length + 1, parameterSyntaxes.Count);
+                foreach (var (verifier, paramSyntax) in paramVerifiers.Append(returnTypeVerifier).Zip(parameterSyntaxes, (v, p) => (v, p)))
+                {
+                    var type = model.GetTypeInfo(paramSyntax.Type!).Type!;
+                    Assert.NotNull(type);
+                    verifier(type);
```

Comment:
>verifier [](start = 20, length = 8)

Why do we need to call verifier again? Would it be sufficient to assert that the type is equal to the one from the symbol? Same for ConvertedTo type.